transportation_splitter.py

```diff
@@ -331,50 +361,50 @@ def apply_lr_scope(
             if not is_applicable:
                 return None
             if new_lr:
-                output[LR_SCOPE_KEY] = new_lr
+                output_dict[LR_SCOPE_KEY] = new_lr
 
         for key in x.keys():
             if key == LR_SCOPE_KEY:
                 continue
-            clean_sub_prop = apply_lr_scope(x[key], split_segment, original_segment_length, f"{xpath}.{key}")
+            clean_sub_prop = apply_lr_scope(x[key], split_segment, original_segment_length, min_overlapping_length_meters, f"{xpath}.{key}")
             if clean_sub_prop is not None:
-                output[key] = clean_sub_prop
+                output_dict[key] = clean_sub_prop
 
-        return output if output else None
+        return output_dict if output_dict else None
     else:
         return x
 
-def get_trs(turn_restrictions, connector_ids):
+def get_trs(turn_restrictions, connectors: list[dict]):
     # extract TR references structure;
     # this will be used after split is complete to identify for each segment_id reference which of the splits of the original segment_id to use;
-    # this step includes pruning out the TRs that don't apply for this split - we check that the TR's first connector id appears in the correct index in connector_ids corresponding to the TR's heading scope (for forward: index=1, for backward: index=0)
+    # this step includes pruning out the TRs that don't apply for this split - we check that the TR's first connector id appears in the correct index in connectors corresponding to the TR's heading scope (for forward: index=1, for backward: index=0)
     if turn_restrictions is None:
         return None, None
 
     flattened_tr_seq_items = []
-    trs_to_keep = []
+    trs_to_keep: list[dict] = []
     for tr in turn_restrictions:
         tr_heading = (tr.get("when") or {}).get("heading")
         tr_sequence = tr.get("sequence")
         if not tr_sequence or len(tr_sequence) == 0:
             continue
 
-        if not connector_ids or len(connector_ids) != 2:
+        if not connectors or len(connectors) != 2:
             # at this point modified segments are expected to have exactly two connector ids, skip edge cases that don't
```

Comment:
We should check this early in the process - if turn restrictions are included in the config, split at connectors must be enabled as well

Reply:
Could add checking for valid configs as a future feature/issue if you want